Adding issue template suggestions (#2416)
inmanta/inmanta-core: .github/ISSUE_TEMPLATE/enhancement.md

> --

Reviewer: ```suggestion
---
```

name: Enhancement
about: Suggest an idea for this project
title: ''
labels: enhancement
assignees: ''

---

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

**Describe the solution you'd like**
A clear and concise description of what you want to happen.

**Describe alternatives you've considered**
A clear and concise description of any alternative solutions or features you've considered.

**Additional context**

> Add any other context or screenshots about the feature request here.

Reviewer: I am wondering whether a section _Additional context_ is useful. The goal of the template is to guide the creator of the ticket as such that he/she doesn't forget to mention necessary information. I am afraid that this section doesn't provide much guidance.

Author: Removed it in https://github.com/inmanta/inmanta/pull/2416/commits/4f10b989b54c0c76c2a13c81416026cef61a37ed

---
name: Enhancement
about: Suggest an idea for this project
title: ''
labels: enhancement
assignees: ''

---

**Self checklist**

 - [ ] I checked that this issue hasn't been created before
 - [ ] I added any relevant label to the issue

> # Describe the solution you'd like

Reviewer 2: This is weird title to see in a feature request. 

```suggestion
# Description
```

Author: Changed it in https://github.com/inmanta/inmanta/pull/2416/commits/5714de4776cf72764d9697480f3502ed2ae70800

> A clear and concise description of what you want to happen.

Reviewer 2: I think an example text would be more instructive. 

(i.e. write out a specific enhancement request)

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/5714de4776cf72764d9697480f3502ed2ae70800

> # Is your feature request related to a problem? Please describe.

Reviewer 2: ```suggestion
# Context
```

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/5714de4776cf72764d9697480f3502ed2ae70800

> A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

Reviewer 2: I think that expanding the example a bit and removing the rest would be beter?

@bart: wdyt?

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/5714de4776cf72764d9697480f3502ed2ae70800

---
name: Enhancement
about: Suggest an idea for this project
title: ''
labels: enhancement
assignees: ''

---

**Self checklist**

 - [ ] This issue hasn't been already created
 - [ ] Everything is sound and clear, all sections are filled-up
 - [ ] Relevant labels have been added

# Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

> # Describe the solution you'd like

Reviewer 2: I would add this description on top

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/5090b071922f322cee876f9d6288b7279514d3b2

---

> name: Enhancement

Reviewer 3: Is there a reason to not call this "Feature request"? Which imho is a well defined, well known term.

> about: Suggest an idea for this project

Reviewer 3: ```suggestion
about: Suggest a feature or enhancement request for this project
```

title: ''
labels: enhancement
assignees: ''

---

> 

Reviewer 3: This is the default github feature request template which is actually quite clear and again really tries to guide the user:

```
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

**Describe the solution you'd like**
A clear and concise description of what you want to happen.

**Describe alternatives you've considered**
A clear and concise description of any alternative solutions or features you've considered.
```

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

**Describe the solution you'd like**
A clear and concise description of what you want to happen.

**Describe alternatives you've considered**
A clear and concise description of any alternative solutions or features you've considered.

Author: This is actually what I initially proposed, see my first commit: https://raw.githubusercontent.com/inmanta/inmanta/156f535111d742230e192df01e62fc7ed15c882a/.github/ISSUE_TEMPLATE/enhancement.md

(Except for the additional context related part)

Reviewer 3: ok

> # Description of the feature

Reviewer 3: I dislike the inconsistency:
- template is called "Enhancement"
- the "about" shown says:  Suggest an idea 
- This says: feature

> (Describe what you want to see change, and why this is a good idea)  

Reviewer 3: ```suggestion
Describe what you want to see changed, and why this is a good idea
```

*Ex: As a developer, when using many `if` statements, I would like to be able to chain them using `elif`. The current syntax doesn't support this in a readable way.*

# Describe alternatives you've considered (if any)

> (Have you considered doing without this feature? How?)  

Reviewer 3: ```suggestion
Have you considered doing without this feature? How?
```
`doing without this feature` is a really strange expression.

---
name: Enhancement
about: Suggest an idea for this project
title: ''
labels: enhancement
assignees: ''

---

> # Context

Reviewer 2: Why does this one have examples and the other one doesn't?

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/47649204fd963ba209a30fd76b240efa0dab2616

> (Is your feature request related to a problem?)  

Reviewer 2: ```suggestion
(Describe what you want to change,  and why this is a good idea)  
```

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/2e2f1f6a155d159d30daa2953c86835f4bde579f

> *Ex: I get frustrated when having to use nested `if` conditions where a `elif` could be used*

Reviewer 2: No frustration here.

I would like this have a form of a user story. 

```suggestion
*Ex: As a developer, when using many if statements, I would like to be able to chain them using elif. The current syntax doesn't support this in a readable way.
```

Author: Changed it in https://github.com/inmanta/inmanta/pull/2416/commits/2e2f1f6a155d159d30daa2953c86835f4bde579f

> # Description of the feature

Reviewer 2: How is this different from context?

Author: I merged the two of them in https://github.com/inmanta/inmanta/pull/2416/commits/2e2f1f6a155d159d30daa2953c86835f4bde579f

(What do you want to see happening)  
*Ex: I would like the keyword `elif` to be added to Inmanta Language*

> # Describe alternatives you've considered

Reviewer 2: I don't know if this is appropriate. It invites a deeper discussion, but it also greatly increases the effort required to make a request?

Author: Should I add "if any"? To make it feel less like a requirement

Author: (Did it in https://github.com/inmanta/inmanta/pull/2416/commits/2e2f1f6a155d159d30daa2953c86835f4bde579f)

(Have you considered doing without this feature? How?)  
*Ex: The same behavior can be achieved currently doing something like:*
```
if condition1:
    ...
else:
    if contition2:
    ...
    end
end
```

> # (Optional) Are you willing to contribute and try implementing this feature?

Reviewer 2: I would remove this section.

Author: Removed it in https://github.com/inmanta/inmanta/pull/2416/commits/2e2f1f6a155d159d30daa2953c86835f4bde579f
